src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2699,6 +2718,25 @@ internal void DoCommand(EventCommandEventArgs commandArgs)
 
                         // There is a good chance EnabledForAnyListener are not as accurate as
                         // they could be, go ahead and get a better estimate.
+
+                        // For self-describing events, loop through the dispatchers m_Level and m_Keywords to compute
+                        // the remaining listeners' maximum level and keywords.
+                        if (SelfDescribingEvents)
```

Comment:
I think its legal to create an EventSource using manifested types and then call Write<T>(...) to do one-off serialization of self-describing events. In this case SelfDescribingEvents would be false, but it would still be necessary to track level/keywords.

(If you are curious I only spotted this because the code above that set level and keywords didn't check SelfDescribingEvents and I noticed the invariant wasn't consistent)